Gitweb:     http://git.kernel.org/git/?p=linux/kernel/git/torvalds/linux-2.6.git;a=commit;h=58607b30fc0f2230a189500112c7a7cca02804cf
Commit:     58607b30fc0f2230a189500112c7a7cca02804cf
Parent:     5db524bd469290212781e1cb0293b57a7a1c000a
Author:     Larry Finger <Larry.Finger@lwfinger.net>
AuthorDate: Sat Jan 3 23:28:25 2009 -0600
Committer:  James Bottomley <James.Bottomley@HansenPartnership.com>
CommitDate: Mon Jan 5 12:17:42 2009 -0600

    [SCSI] aha152x_cs: Fix regression that keeps driver from using shared interrupts
    
    At some point since 2.6.22, the aha152x_cs driver stopped working and
    started erring on load with the following messages:
    
    kernel: pcmcia: request for exclusive IRQ could not be fulfilled.
    kernel: pcmcia: the driver needs updating to supported shared IRQ lines.
    
    With the following change, the driver works with shared IRQs.
    
    Signed-off-by: Larry Finger <Larry.Finger@lwfinger.net>
    Cc: Stable <stable@kernel.org>		[2.6.28], [2.6.27], [2.6.26]
    Signed-off-by: James Bottomley <James.Bottomley@HansenPartnership.com>

 drivers/scsi/pcmcia/aha152x_stub.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/drivers/scsi/pcmcia/aha152x_stub.c b/drivers/scsi/pcmcia/aha152x_stub.c
index 165ff88..67cde01 100644
--- a/drivers/scsi/pcmcia/aha152x_stub.c
+++ b/drivers/scsi/pcmcia/aha152x_stub.c
@@ -114,7 +114,7 @@ static int aha152x_probe(struct pcmcia_device *link)
     link->io.NumPorts1 = 0x20;
     link->io.Attributes1 = IO_DATA_PATH_WIDTH_AUTO;
     link->io.IOAddrLines = 10;
-    link->irq.Attributes = IRQ_TYPE_EXCLUSIVE;
+    link->irq.Attributes = IRQ_TYPE_DYNAMIC_SHARING;
     link->irq.IRQInfo1 = IRQ_LEVEL_ID;
     link->conf.Attributes = CONF_ENABLE_IRQ;
     link->conf.IntType = INT_MEMORY_AND_IO;
